lib/candles/abstract.js

```diff
@@ -158,7 +158,7 @@ function AbstractCandles(client) {
 	};
 
 	this.validData = function (data) {
-		if (_.size(data) > 0) {
+		if (Array.isArray(data) && data.length > 0) {
 			return _.first(data)[self.validationKey];
 		}
 		return true;
```

Comment:
Why it returns `true` for invalid data?

Reply:
That's a good question. I've tried to "fix" it and it made things only worse. It can be written better way but it means more serious refactoring.
Also, notice that returning `true` here breaks the async loop, which is good assuming we don't want to retry on a request with an invalid data. The procedure assumes that everything that fails leads to retry, except the situation mentioned above.

Reply:
Ok, so leave it as is. :)